modules/core/src/main/java/org/locationtech/jts/index/strtree/STRtree.java

```diff
@@ -373,5 +402,91 @@ public Object nearestNeighbour(Envelope env, Object item, ItemDistance itemDist)
           ((ItemBoundable) minPair.getBoundable(1)).getItem()
       };
   }
+  private Object[] nearestNeighbour(BoundablePair initBndPair, double maxDistance, int k) 
```

Comment:
Might work. I resolve the overload by runtime inspection of the number value using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger. (yikes...)